qpid-jms-client/src/main/java/org/apache/qpid/jms/JmsConnectionFactory.java

```diff
@@ -591,6 +613,14 @@ public long getConnectTimeout() {
         return this.connectTimeout;
     }
 
+    public void setCompletionThreads(final int completionThreads) {
```

Comment:
> EDIT: actually, is it? I'm not seeing where it would actually share?

The completion factory create a singleton instance of `sharedRefCnt` of `ForkJoinPool` that allows sharing the same FJ pool unless every connection that reference it, got closed. If that happen, the last one would dispose it, leaving incoming connections (if any) able to create a new one, similarly to the shared event loop group of #45

Reply:
> That would introduce a similar issue to the other JIRA/PR, where people might have different configurations that then needs different pools here...but that seems like it would be easily solved in a pretty similar way, effectively just a map of different setups to match against.

This can be done creating some "leaky" (by purpose) singleton shared map with per-configuration pool, but is it a valid use case?
Or the typical use case is to have a single factory per-application?

Reply:
I've seen both approaches. Neither is unusual. It really just depends what they are doing.

E.g if they are configuring ClientID's on the factory, they can necessarily only create 1 connection [at a time] with that factory, and might have more than one. Different frameworks, different config.

I would typically expect them to use the same config for options like this even if they do...but, really nothing requires they do that, and it may not always be in their control that they do, so it should work if they dont. Same as comments on the other PR, really.